packages/remix-server-runtime/build.ts

```diff
@@ -15,7 +15,7 @@ export interface ServerBuild {
   publicPath: string;
   assetsBuildDirectory: string;
   future: FutureConfig;
-  dev?: { port: number };
```

Comment:
The intention is to allow omission of `dev`, not to let it be explicitly set to `undefined`. From what I can tell, including `| undefined` is not the semantics we want here.

Where is `dev` getting set to `undefined` explicitly? I think a better solution would be to make sure its omitted there instead.

Reply:
The problem originates here:

https://github.com/remix-run/remix/blob/3e589152bc717d04e2054c31bea5a1056080d4b9/packages/remix-dev/server-build.ts#L14

Because `ServerBuild["dev"]` is an optional property, the type of `dev` must be widened to include `undefined`. You could change this type to `NonNullable<ServerBuild["dev"]>`, but that seems like it might be less correct than the current change.